### PR TITLE
test: added tests for notifications reducer

### PR DESCRIPTION
**Pull request by jglover**
Added tests for notifications reducer. This revealed a few problems with the reducer that I'll address in an issue.

### CODE REVIEW

File: __tests__/tests/reducers/notifications.reducer.js [hunk 1]
```diff
@@ -0,0 +1,475 @@
+import React from 'react';
```
Comment (alejandronanez) [inline __tests__/tests/reducers/notifications.reducer.js]: Generally speaking, I agree with @andrewda about not nesting describes, that can get quite messy pretty quickly.
I think one describe per test suite is fine and just using `it` clauses should be enough for describing a good test suite. Can you please flatten your tests so you only use 1 describe and multiple `it` statements?

Comment (jglover) [reply __tests__/tests/reducers/notifications.reducer.js]: Sure

File: __tests__/tests/components/EntityInfo.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
 import { shallow } from 'enzyme';
 
-import organization from '../../data/organization';
-import user from '../../data/user';
+import organization from '../../data/api/organization';
```
[the remark this answers is not in the document]
Comment (jglover) [reply __tests__/tests/components/EntityInfo.js]: This occurred to me, I'll update it.

File: __tests__/tests/reducers/notifications.reducer.js [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+  GET_ALL_NOTIFICATIONS,
+  MARK_NOTIFICATION_AS_READ,
+  MARK_ALL_NOTIFICATIONS_AS_READ,
+  GET_NOTIFICATIONS_COUNT,
+  MARK_REPO_AS_READ,
+} from 'notifications/notifications.type';
+import { notification } from '../../data/api/notification';
+import { authError } from '../../data/api/error';
+
+describe('Notifications reducer', () => {
```
Comment (andrewda) [inline __tests__/tests/reducers/notifications.reducer.js]: Use title case: "Notifications Reducer".

**Comment by jglover** [comment]
@alejandronanez @andrewda The requested changes have been made :)

File: __tests__/tests/components/EntityInfo.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (alejandronanez) [reply __tests__/tests/components/EntityInfo.js]: Hmmm we're still using relative imports here.

File: __tests__/tests/components/StateBadge.js [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 import { shallow } from 'enzyme';
 
-import { closed as closedIssue, open as openIssue } from '../../data/issue';
+import { closed as closedIssue, open as openIssue } from '../../data/api/issue';
 import {
   closed as closedPr,
   open as openPr,
   merged as mergedPr,
-} from '../../data/pull-request';
+} from '../../data/api/pull-request';
```
Comment (alejandronanez) [inline __tests__/tests/components/StateBadge.js]: Also we're using relative imports here.

Comment (jglover) [reply __tests__/tests/components/StateBadge.js]: These are just tests I touched when I moved the test data files, my discipline tells me update all tests in a different PR but I'll do it here.

Comment (jglover) [reply __tests__/tests/components/StateBadge.js]: @alejandronanez all references updated to use alias instead of absolute paths.

**Comment by jglover** [comment]
@alejandronanez updated paths to use alias instead of absolute paths across all tests.

**Comment by andrewda** [comment]
Thanks!

**Pull request merged**